support/sys-socket.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * <sys/socket.h> wrapper functions.
+ *
+ * Authors:
+ *   Steffen Kiess (s-kiess@web.de)
+ *
+ * Copyright (C) 2015 Steffen Kiess
+ */
+
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <sys/un.h>
+
+#include <stddef.h>
+
+#include "map.h"
+#include "mph.h"
+
+G_BEGIN_DECLS
+
+int
+Mono_Posix_Syscall_socketpair (int domain, int type, int protocol, int* sv0, int* sv1)
```

Comment:
> Which overload? I'd still argue for the out parameters (readability!), but I can be convinced otherwise. :-)

Ok, I've removed the int[] overload. I've called the parameters socket1 and socket2 (unlike pipe() there is no difference between the two file descriptors returned by socketpair().)